### PR TITLE
Reva based simple ocis

### PR DESCRIPTION
**Pull request by butonic**
This PR uses reva and the simple build flag to configure a system that shows how the oidc auth works and how extensions talk to phoenix.
- [ ] [ocis-konnectd](https://github.com/owncloud/ocis-konnectd/pull/6) is not yet ready, so I am using revas oidc provider
- [x] ~requires https://github.com/owncloud/ocis-phoenix/pull/25 to fix config setting for hello.js~ has been merged
- [x] ran into problems with configuring extensions from ocis, tracked in https://github.com/owncloud/ocis/issues/81 setting env vars as a workaround, they will be picked up when micro starts the sub processes
  - [x] only set them if they are not set, so they can be overridden ... or find out how to properly fix the bug

- [ ] I am still seeing an error when pressing submit in hello: `Saving your name failed
Missing required parameter: body` ... maybe @LukasHirt can give me a hint ... tracked in https://github.com/owncloud/ocis-hello/issues/41

### CODE REVIEW

**Comment by tboerger** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/reva_simple.go  1
- pkg/command/reva_ocis.go  1
         

Clones added
============
- pkg/command/reva_simple.go  8
- pkg/command/reva_ocis.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=4803507&bid=15983602)

File: pkg/command/server_simple.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 
 var (
 	// SimpleRuntimeServices declares which services will be started for the fullstack server
-	SimpleRuntimeServices = []string{"hello", "konnectd", "phoenix"}
+	SimpleRuntimeServices = []string{"hello", "konnectd", "phoenix", "reva"}
```
Comment (butonic) [inline pkg/command/server_simple.go]: hm konnect is not necessary in this case

**Comment by tboerger** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/phoenix_simple.go  2
- pkg/command/reva_ocis.go  1
- pkg/command/reva_simple.go  1
         

Clones added
============
- pkg/command/reva_ocis.go  1
- pkg/command/reva_simple.go  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=4803507&bid=15983602)

**Comment by LukasHirt** [comment]
> I am still seeing an error when pressing submit in hello: Saving your name failed Missing required parameter: body ... maybe @LukasHirt can give me a hint ... tracked in

I'll take care.

**Comment by tboerger** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/phoenix_simple.go  2
- pkg/command/reva_simple.go  1
- pkg/command/reva_ocis.go  1
         

Clones added
============
- pkg/command/reva_simple.go  8
- pkg/command/reva_ocis.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=4803507&bid=15983602)

**Comment by tboerger** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/reva_simple.go  1
- pkg/command/phoenix_simple.go  2
- pkg/command/phoenix_ocis.go  3
- pkg/command/reva_ocis.go  1
         

Clones added
============
- pkg/command/reva_simple.go  19
- pkg/command/reva_ocis.go  71
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=4803507&bid=15983602)

File: pkg/command/phoenix_simple.go [hunk 2]
```diff
@@ -14,12 +16,23 @@ func configurePhoenix(cfg *config.Config) *svcconfig.Config {
 	cfg.Phoenix.Tracing.Enabled = false
 	cfg.Phoenix.HTTP.Addr = "localhost:9100"
 	cfg.Phoenix.HTTP.Root = "/"
```
Comment (refs) [inline pkg/command/phoenix_simple.go]: I'd get rid of these as they're being set by the Viper defaults

**Comment by refs** [comment]
works on my machine ®️

**Comment by tboerger** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/reva_simple.go  1
- pkg/command/reva_ocis.go  1
- pkg/command/phoenix_ocis.go  3
- pkg/command/phoenix_simple.go  2
         

Clones added
============
- pkg/command/reva_simple.go  19
- pkg/command/reva_ocis.go  71
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=4803507&bid=15983602)

**Comment by tboerger** [comment]
Why does such poorly formatted code get merged at all?

Beside that, using setenv is so much wrong...

**Pull request merged**